services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_specs.py

```diff
@@ -133,13 +136,30 @@ def _update_resource_limits_and_reservations(
             resources["limits"] = limits
             deploy["resources"] = resources
             spec["deploy"] = deploy
+
+            nano_cpu_limits = limits["cpus"] * 10**9
+            mem_limits = limits["memory"]
         else:
             # compos spec version 2
             spec["mem_limit"] = f"{memory.limit}"
             spec["mem_reservation"] = f"{memory.reservation}"
             # NOTE: there is no distinction between limit and reservation, taking the higher value
             spec["cpus"] = float(max(cpu.limit, cpu.reservation))
 
+            nano_cpu_limits = spec["cpus"] * 10**9
```

Comment:
```python
 nano_cpu_limits = limits["cpus"] * 10**9
mem_limits = spec["mem_limit"]
```
are repeated in both options of if. Just put it outside?

Reply:
done